clang -Werror: Kokkos_FixedBufferMemoryPool.hpp:140:28: error: unused parameter 'alloc_size'

clang/3.8.1, clang/4.0.1 Serial and Pthread and clang/8 nightly builds are failing with `-Werror` enabled with this message:

```
/home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_clang_381/kokkos/core/src/impl/Kokkos_FixedBufferMemoryPool.hpp:140:28: error: unused parameter 'alloc_size' [-Werror,-Wunused-parameter]
```

Reproducer sample:
```
 #   Load modules:
        module load sems-env sems-cmake/3.12.2 kokkos-env kokkos-hwloc/1.10.1/base sems-clang/3.8.1

  #   Use generate_makefile line below to call cmake which generates makefile for this build:
        <KOKKOS_PATH>/generate_makefile.bash --with-devices=Serial --arch=Kepler35 --compiler=/projects/sems/install/rhel6-x86_64/sems/compiler/clang/3.8.1/base/bin/clang++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="11" --ldflags=""  --kokkos-path=<KOKKOS_PATH> --no-examples --with-options=enable_large_mem_tests
```

--- a/core/src/impl/Kokkos_FixedBufferMemoryPool.hpp
+++ b/core/src/impl/Kokkos_FixedBufferMemoryPool.hpp
@@ -138,6 +138,7 @@ class FixedBlockSizeMemoryPool
 
   KOKKOS_INLINE_FUNCTION
   void* allocate(size_type alloc_size) const noexcept {
+    (void)alloc_size;
     KOKKOS_EXPECTS(alloc_size <= Size);
     auto free_idx_counter = Kokkos::atomic_fetch_add(
         (volatile size_type*)&m_first_free_idx, size_type(1));

